src/backend/renderer/pixman/mod.rs

```diff
@@ -0,0 +1,1183 @@
+
+        let mut clip_region =
+            pixman::Region32::init_rect(0, 0, self.output_size.w as u32, self.output_size.h as u32);
+
+        let damage_boxes = damage
+            .iter()
+            .copied()
+            .map(|mut rect| {
+                rect.loc += dst.loc;
+
+                let rect = self.transform.transform_rect_in(rect, &self.size);
+
+                let p1 = rect.loc;
+                let p2 = p1 + rect.size.to_point();
+                pixman::Box32 {
+                    x1: p1.x,
+                    y1: p1.y,
+                    x2: p2.x,
+                    y2: p2.y,
+                }
+            })
+            .collect::<Vec<_>>();
+        let damage_region = pixman::Region32::init_rects(&damage_boxes);
+        clip_region = clip_region.intersect(&damage_region);
+
+        target_image.set_clip_region32(Some(&clip_region))?;
+
+        target_image.composite32(
+            op,
+            &solid,
+            None,
+            (0, 0),
+            (0, 0),
+            (0, 0),
+            (target_image.width() as i32, target_image.height() as i32),
+        );
+
+        if debug.contains(DebugFlags::TINT) {
+            target_image.composite32(
+                Operation::Over,
+                &self.renderer.tint,
+                None,
+                (0, 0),
+                (0, 0),
+                (0, 0),
+                (target_image.width() as i32, target_image.height() as i32),
+            );
+        }
+
+        target_image.set_clip_region32(None)?;
+
+        Ok(())
+    }
+}
+
+impl<'frame> Frame for PixmanFrame<'frame> {
+    type Error = PixmanError;
+
+    type TextureId = PixmanTexture;
+
+    fn id(&self) -> usize {
+        0
+    }
+
+    #[profiling::function]
+    fn clear(&mut self, color: [f32; 4], at: &[Rectangle<i32, Physical>]) -> Result<(), Self::Error> {
+        self.draw_solid_color(
+            Rectangle::from_loc_and_size((0, 0), self.size),
+            at,
+            color,
+            Operation::Src,
+            DebugFlags::empty(),
+        )
+    }
+
+    #[profiling::function]
+    fn draw_solid(
+        &mut self,
+        dst: Rectangle<i32, Physical>,
+        damage: &[Rectangle<i32, Physical>],
+        color: [f32; 4],
+    ) -> Result<(), Self::Error> {
+        let op = if color[3] == 1f32 {
+            Operation::Src
+        } else {
+            Operation::Over
+        };
+        self.draw_solid_color(dst, damage, color, op, self.renderer.debug_flags)
+    }
+
+    #[profiling::function]
+    fn render_texture_from_to(
+        &mut self,
+        texture: &Self::TextureId,
+        src: Rectangle<f64, BufferCoords>,
+        dst: Rectangle<i32, Physical>,
+        damage: &[Rectangle<i32, Physical>],
+        src_transform: Transform,
+        alpha: f32,
+    ) -> Result<(), Self::Error> {
+        let Some(target_image) = self.renderer.target.as_ref().map(|target| target.image()) else {
+            return Ok(());
```

Comment:
Why is this no error?

Reply:
I think I had that as an error, but removed it to align the behavior with the gles renderer which does not error out in this case either.

Reply:
I think for the GlesRenderer this is the case, as we can have an implicit framebuffer (EGLSurface). For this one there is no default target, right?

Reply:
Ah, yeah, good point. I will reintroduce the error then

Reply:
Added the error back